packages/@aws-cdk/aws-applicationautoscaling/lib/scalable-target.ts

```diff
@@ -0,0 +1,211 @@
+   */
+  minCapacity: number;
+
+  /**
+   * The maximum value that Application Auto Scaling can use to scale a target during a scaling activity.
+   */
+  maxCapacity: number;
+
+  /**
+   * Role that allows Application Auto Scaling to modify your scalable target.
+   *
+   * @default A role is automatically created
+   */
+  role?: iam.IRole;
+
+  /**
+   * The resource identifier to associate with this scalable target.
+   *
+   * This string consists of the resource type and unique identifier.
+   *
+   * @example service/ecsStack-MyECSCluster-AB12CDE3F4GH/ecsStack-MyECSService-AB12CDE3F4GH
+   * @see https://docs.aws.amazon.com/autoscaling/application/APIReference/API_RegisterScalableTarget.html
+   */
+  resourceId: string;
+
+  /**
+   * The scalable dimension that's associated with the scalable target.
+   *
+   * Specify the service namespace, resource type, and scaling property.
+   *
+   * @example ecs:service:DesiredCount
+   * @see https://docs.aws.amazon.com/autoscaling/application/APIReference/API_ScalingPolicy.html
+   */
+  scalableDimension: string;
+
+  /**
+   * The namespace of the AWS service that provides the resource or
+   * custom-resource for a resource provided by your own application or
+   * service.
+   *
+   * For valid AWS service namespace values, see the RegisterScalableTarget
+   * action in the Application Auto Scaling API Reference.
+   *
+   * @see https://docs.aws.amazon.com/autoscaling/application/APIReference/API_RegisterScalableTarget.html
+   */
+  serviceNamespace: string;
+}
+
+/**
+ * Define a scalable target
+ */
+export class ScalableTarget extends cdk.Construct {
+  /**
+   * ID of the Scalable Target
+   *
+   * @example service/ecsStack-MyECSCluster-AB12CDE3F4GH/ecsStack-MyECSService-AB12CDE3F4GH|ecs:service:DesiredCount|ecs
+   */
+  public readonly scalableTargetId: string;
+
+  /**
+   * The role used to give AutoScaling permissions to your resource
+   */
+  public readonly role: iam.IRole;
+
+  private readonly actions = new Array<cloudformation.ScalableTargetResource.ScheduledActionProperty>();
+
+  constructor(parent: cdk.Construct, id: string, props: ScalableTargetProps) {
+    super(parent, id);
+
+    if (props.maxCapacity < 0) {
+      throw new RangeError(`maxCapacity cannot be negative, got: ${props.maxCapacity}`);
+    }
+    if (props.minCapacity < 0) {
+      throw new RangeError(`minCapacity cannot be negative, got: ${props.minCapacity}`);
+    }
+    if (props.maxCapacity < props.minCapacity) {
+      throw new RangeError(`minCapacity (${props.minCapacity}) should be lower than maxCapacity (${props.maxCapacity})`);
+    }
+
+    this.role = props.role || new iam.Role(this, 'Role', {
+      assumedBy: new iam.ServicePrincipal('application-autoscaling.amazonaws.com')
+    });
+
+    const resource = new cloudformation.ScalableTargetResource(this, 'Resource', {
+      maxCapacity: new cdk.Token(() => props.maxCapacity),
```

Comment:
Can you not `new cdk.Token(props.maxCapacity)`? What's the point of wrapping in a function?

Reply:
At some point this property was mutable, but it no longer is. Don't even need the Token anymore.